From 64363d6802851fba783d896300f5d713b8433d58 Mon Sep 17 00:00:00 2001
Message-Id: <64363d6802851fba783d896300f5d713b8433d58.1693597912.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1693597912.git.sharlatanus@gmail.com>
References: <cover.1693597912.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 6 Aug 2023 14:33:31 +0100
Subject: [PATCH v2 2/9] gnu: Add go-github-com-bitly-timer-metrics.

* gnu/packages/golang.scm (go-github-com-bitly-timer-metrics): New variable.
---
 gnu/packages/golang.scm | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index 70d5af6107..b569c9edaa 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -2137,6 +2137,29 @@ (define-public go-github-com-bitly-go-hostpool
 robin or epsilon greedy mode, and unresponsive hosts are avoided.")
     (license license:expat)))
 
+(define-public go-github-com-bitly-timer-metrics
+  (package
+    (name "go-github-com-bitly-timer-metrics")
+    (version "1.0.0")
+    (source (origin
+              (method git-fetch)
+              (uri (git-reference
+                    (url "https://github.com/bitly/timer_metrics")
+                    (commit (string-append "v" version))))
+              (file-name (git-file-name name version))
+              (sha256
+               (base32
+                "02fhx8hx8126m2cgxw9fm8q2401r7zfann8b5zy5yyark1sgkrb4"))))
+    (build-system go-build-system)
+    (arguments
+     '(#:import-path "github.com/bitly/timer_metrics"))
+    (home-page "https://github.com/bitly/timer_metrics")
+    (synopsis "Capture timings and enable periodic metrics every @var{n} events")
+    (description
+     "This package provides an efficient way to capture timing information and
+periodically output metrics")
+    (license license:expat)))
+
 (define-public go-github-com-blanu-dust
   (package
     (name "go-github-com-blanu-dust")
-- 
2.40.1

